core/plugins/org.polarsys.capella.core.sirius.analysis/src/org/polarsys/capella/core/sirius/analysis/TitleBlockServices.java

```diff
@@ -0,0 +1,643 @@
+      annotationCol.getDetails().put(LINE, String.valueOf(lineNumber));
+      annotationCol.getDetails().put(COLUMN, String.valueOf(j));
+      annotationCol.getDetails().put(IS_ELEMENT_TITLE_BLOCK, TRUE);
+      annotationCols.add(annotationCol);
+      representation.getEAnnotations().add(annotationCol);
+    }
+    annotationLine.getReferences().addAll(annotationCols);
+    representation.getEAnnotations().add(annotationLine);
+  }
+
+  public void createTitleBlockColumn(EObject titleBlock, EObject diagram) {
+    if (!isDiagramTitleBlock((DAnnotation) titleBlock)
+        || "True".equals(((DAnnotation) titleBlock).getDetails().get(IS_ELEMENT_TITLE_BLOCK))) {
+      int columnNumber = Integer.valueOf(((DAnnotation) titleBlock).getDetails().get(COLUMN));
+      titleBlock = getParentTitleBlock((DAnnotation) titleBlock, diagram);
+      DRepresentation representation = null;
+      if ((diagram instanceof DRepresentation)) {
+        representation = (DRepresentation) diagram;
+      }
+      if (representation != null) {
+        if (titleBlock instanceof DAnnotation) {
+          int numLines = ((DAnnotation) titleBlock).getReferences().size();
+          List<EObject> lines = ((DAnnotation) titleBlock).getReferences();
+          for (int i = 0; i < numLines; i++) {
+            DAnnotation annotationCol = DescriptionFactory.eINSTANCE.createDAnnotation();
+            annotationCol.setSource(SOURCE_CELL_PREFIX);
+            annotationCol.getDetails().put(NAME, "");
+            annotationCol.getDetails().put(CONTENT, "");
+            annotationCol.getDetails().put(LINE, String.valueOf(i - 1));
+            annotationCol.getDetails().put(COLUMN, String.valueOf(columnNumber));
+            annotationCol.getDetails().put(IS_ELEMENT_TITLE_BLOCK, TRUE);
+            if (lines.get(i) instanceof DAnnotation) {
+              ((DAnnotation) (lines.get(i))).getReferences().add(columnNumber + 1, annotationCol);
+              representation.getEAnnotations().add(annotationCol);
+            }
+          }
+        }
+      }
+      renumberColumns((DAnnotation) titleBlock);
+    }
+  }
+
+  private void renumberColumns(DAnnotation titleBlock) {
+    int lineNumber = -1;
+    for (EObject reference : titleBlock.getReferences()) {
+      if (reference instanceof DAnnotation) {
+        int columnNumber = 0;
+        for (EObject ref : ((DAnnotation) reference).getReferences()) {
+          ((DAnnotation) ref).getDetails().put(LINE, String.valueOf(lineNumber));
+          ((DAnnotation) ref).getDetails().put(COLUMN, String.valueOf(columnNumber));
+          columnNumber++;
+        }
+      }
+      lineNumber++;
+    }
+  }
+
+  public int getNumLines(DAnnotation titleBlock) {
+    if (isDiagramTitleBlock(titleBlock)) {
+      return titleBlock.getReferences().size();
+    }
+    return titleBlock.getReferences().size() - 1;
+  }
+
+  public int getNumColumns(DAnnotation titleBlock) {
+    if (titleBlock.getReferences().size() > 1) {

```

Comment:
this function works only for Element TB. For diagram TB is not used?

Reply:
This is because you cannot add lines or columns to diagram title block.

Reply:
Renamed.